src/mapml/features/feature.js

```diff
@@ -0,0 +1,270 @@
+/**
+ * M.Feature is a extension of L.Path that understands mapml feature markup
+ * It converts the markup to the following structure (abstract enough to encompass all feature types) for example:
+ *  this._outlinePath = HTMLElement;
+ *  this._parts = [
+ *    {
+ *      path: HTMLElement,
+ *      rings:[
+ *        {points:[{x:1,y:1}, ...]},
+ *        ...
+ *      ],
+ *      subrings:[
+ *        {points:[{x:2, y:2}, ...], cls:"Span Class Name", path: HTMLElement,},
+ *        ...
+ *      ],
+ *      cls:"className",
+ *    },
+ *    ...
+ *  ];
+ */
+export var Feature = L.Path.extend({
+  options: {
+    accessibleTitle: "Feature",
+  },
+
+  /**
+   * Initializes the M.Feature
+   * @param {HTMLElement} markup - The markup representation of the feature
+   * @param {Object} options - The options of the feature
+   */
+  initialize: function (markup, options) {
+    this.type = markup.tagName.toUpperCase();
+    L.setOptions(this, options);
+
+    this._createGroup();  // creates the <g> element for the feature, or sets the one passed in options as the <g>
+
+    this._parts = [];
+    this._markup = markup;
+    this.options.zoom = markup.getAttribute('zoom') || this.options.nativeZoom;
+
+    this._generateOutlinePoints();
+    this._convertMarkup();
+
+    this.isClosed = this._isClosed();
+  },
+
+  /**
+   * Removes the focus handler, and calls the leaflet L.Path.onRemove
+   */
+  onRemove: function () {
+    L.DomEvent.off(this.group, "keyup keydown mousedown", this._handleFocus, this);
+    L.Path.prototype.onRemove.call(this);
+  },
+
+  /**
+   * Creates the <g> conditionally and also applies event handlers
+   * @private
+   */
+  _createGroup: function(){
+    if(this.options.multiGroup){
+      this.group = this.options.multiGroup;
+    } else {
+      this.group = L.SVG.create('g');
+      this.group.setAttribute('role', 'region');
```

Comment:
Although [`role="region"`](https://www.w3.org/TR/wai-aria/#region) sounds like a good role because these are also geographic regions, that's not what the role is intended for. `region` is one of few [landmark roles](https://www.w3.org/TR/wai-aria/#landmark_roles) that may be exposed to users in an interface such as this:

![document-regions](https://user-images.githubusercontent.com/26493779/112012360-6f2be600-8b29-11eb-8664-a0064d151a1a.png)

(image credit: https://www.youtube.com/watch?v=7Rs3YpsnfoI&t=157s)

Accessibility expert Scott O'Hara [states](https://www.scottohara.me/blog/2018/03/03/landmarks.html#region):
<q>The use of region landmarks should be limited, as too many landmarks can dilute their usefulness in quickly navigating to important areas of a page.</q>

And, in testing `role="region"` also does not seem to expose `aria-expanded="true|false` whereas `role="button"` does. IMO, `role="button"` in conjunction to `aria-roledescription="Feature"` makes the most sense.

Reply:
OTOH maybe it's best to discuss this in a separate issue.

Reply:
It was a recommendation given by @shepazu , it was suggested as a way to enable aria-label on `<g>` elements.